### PR TITLE
Optimize Hitboxes collision testing for "big" objects

### PR DESCRIPTION
**Pull request by victorlevasseur**
"big" = objects with a lot of hitboxes.

# What is the optimization ?

This PR adds a new ```GetHitboxes(sf::FloatRect hint)``` overload to ```RuntimeObject``` to get Hitboxes with a AABB hint. 

However the object is not required to follow the hint: it just needs to return at least the hitboxes intersecting the AABB (the default implementation only forwards the call to the standard ```GetHitboxes()``` method).

This new method is used by ```RuntimeObject::IsCollidingWith``` (so it also works for the Platformer behavior :) ) and ```RuntimeObject::SeparateFromObjects```.

# Which object uses this new overload ?

The only object currently taking advantage of the hint is the ```TileMapObject``` as it contains ```TileCountX``` * ```TileCountY``` hitboxes causing poor performances when using all the hitboxes. This object now get only the hitboxes of the tiles with the AABB hint.

# Performances

Obviously, this brings a **huge** performance improvement when testing the collisions with a TileMapObject or having a TileMapObject as a platform:

 * before this optimization: big FPS drop (~30-50 fps) fps with a 500*300 tilemap and a single colliding object !
 * after this optimization: almost no FPS loss (~180-200 fps) when testing or not the collision between an object and a tilemap (same applies to the PlatformerBehavior)

# A testcase

I made a small testcase to test some scenarios with the new system. You can test the project with the current GDevelop version and this version to see the difference : 
[hitbox-optimization-testcase.zip](https://github.com/4ian/GD/files/425384/hitbox-optimization-testcase.zip)
(forgot to update the images filepath, but they are in the .zip)

### CODE REVIEW

File: Extensions/TileMapObject/RuntimeTileMapObject.cpp [hunk 1]
```diff
@@ -137,6 +137,59 @@ std::vector<Polygon2d> RuntimeTileMapObject::GetHitBoxes() const
...
+    else
+        bottomRight.x = (hint.left + hint.width - GetX()) / GetTileWidth();
+
+    if(hint.top + hint.height > GetY() + GetHeight())
+        bottomRight.y = GetMapHeight() - 1;
+    else
+        bottomRight.y = (hint.top + hint.height - GetY()) / GetTileHeight();
+
+    std::cout << "Getting hitboxes from " << topLeft.x << ";" << topLeft.y << " to ";
+    std::cout << bottomRight.x << ";" << bottomRight.y << std::endl;
```
Comment (4ian) [inline Extensions/TileMapObject/RuntimeTileMapObject.cpp]: @victorlevasseur Should this be removed/commented out? :)

**Comment by victorlevasseur** [comment]
Removed the debug std::cout

**Comment by 4ian** [comment]
Sorry for the delay - looks good to me :) Is it stable and ready to merge?

**Comment by victorlevasseur** [comment]
Just doing some more tests with it to avoid bugs.

**Comment by victorlevasseur** [comment]
No bugs found. Did you review it ?

**Comment by 4ian** [comment]
Did not have the chance to do big tests, but implementation looks good to me!
Feel free to merge it if it's ok for you ;)

**Pull request merged**